rawspec_file.c

```diff
@@ -38,33 +62,32 @@ int open_output_file(const char * dest, const char *stem, int output_idx)
   return fd;
 }
 
-int open_output_file_per_antenna_and_write_header(callback_data_t *cb_data, const char * dest, const char *stem, int output_idx)
+// Open one or more output Filterbank files for the following cases:
+//   * nants = 0
+//   * the set of antennas when nants > 0
+int open_output_file_per_antenna_and_write_header(callback_data_t *cb_data, const char * dest, const char * stem, int output_idx)
 {
   char ant_stem[PATH_MAX+1];
-  if(cb_data->per_ant_out){
-    cb_data->fb_hdr.nchans /= cb_data->Nant;
-  }
-  for(int i = 0; i < (cb_data->per_ant_out ? cb_data->Nant : 1); i++){
-    if(cb_data->per_ant_out){
+    
+  for(int i = 0; i < (cb_data->per_ant_out ? cb_data->Nant : 1); i++) {
+    if(cb_data->per_ant_out) {
       snprintf(ant_stem, PATH_MAX, "%s-ant%03d", stem, i);
     }
-    else{
+    else {
       snprintf(ant_stem, PATH_MAX, "%s", stem);
     }
 
-    cb_data->fd[i] = open_output_file(dest, ant_stem, output_idx);
+    cb_data->fd[i] = open_output_file(cb_data, dest, ant_stem, output_idx, i);
```

Comment:
I reckon the ICS_flag should be a [0,1] flag, so `, i);` -> `, 0);`

Reply:
See earlier remarks about ics flag vs antenna index.